packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts

```diff
@@ -261,6 +309,36 @@ function pickDecorationNodesToRemove(
   return (elements.length > ngDecorators.length) ? ngDecorators : [exprStmt];
 }
 
+// Remove Angular decorators from `Clazz = __decorate([...], Clazz)`, or expression itself if all
+// are removed.
+function pickDecorateNodesToRemove(
+  exprStmt: ts.ExpressionStatement,
+  ngMetadata: ts.Node[],
+  checker: ts.TypeChecker,
+): ts.Node[] {
+
+  const expr = expect<ts.BinaryExpression>(exprStmt.expression, ts.SyntaxKind.BinaryExpression);
+  const callExpr = expect<ts.CallExpression>(expr.right, ts.SyntaxKind.CallExpression);
+  const arrLiteral = expect<ts.ArrayLiteralExpression>(callExpr.arguments[0],
+    ts.SyntaxKind.ArrayLiteralExpression);
+  if (!arrLiteral.elements.every((elem) => elem.kind === ts.SyntaxKind.CallExpression)) {
+    return [];
+  }
+  const elements = arrLiteral.elements as ts.NodeArray<ts.CallExpression>;
+  const ngDecoratorCalls = elements.filter((el) => {
+    if (el.expression.kind !== ts.SyntaxKind.Identifier) {
+      return false;
+    }
+    const id = el.expression as ts.Identifier;
+
+    return identifierIsMetadata(id, ngMetadata, checker);
+  });
+
+  return (elements.length > ngDecoratorCalls.length) ? ngDecoratorCalls : [exprStmt];
```

Comment:
this logic is hard to follow. maybe you want to say:

```
// if all decorators are metadata decorators then return the whole `Class = __decorate([...])'` statement so that it is removed in entirety
return (elements.length === ngDecoratorCalls.length) ? [exprStmt] : ngDecoratorCalls;
```

Reply:
Fixed